### PR TITLE
Generic guns overhaul

### PR DESCRIPTION
**Pull request by Lamandus**
This is a Overhaul of the Generic guns mod. It is more akin to modern BN with its values and balances around 2 different ammo types (JHP and FMJ).
A probably not complete list of changes:
- obsoleted reloaded and AP ammo types (AP was the only option for rifles+)
- added JHP to rifles+
- balanced uncrafting and crafting recipes
- balanced damage output
- updated mergelist with most ammo types present in the game, fixing some bugs (.22 ammo specifically)
- added Sniper ammo and added recipes, mergelists, etc. for it.
- Adding high capacity magazine variants of most rifle types (except anti material)
- adding me as the Maintainer of this mod.

this is tested and working fine as far as I can tell, I will continue testing it more, but it is ready to be merged.

### CODE REVIEW

**Comment by Lamandus** [comment]
sorry, my github desktop was doing something wrong

File: data/mods/Generic_Guns/ammo/pistol.json [hunk 1]
```diff
@@ -2,62 +2,42 @@
...
-    "extend": { "effects": [ "RECYCLED" ] }
-  },
-  {
-    "id": "reloaded_pistol_jhp",
-    "type": "AMMO",
-    "copy-from": "pistol_jhp",
-    "name": { "str_sp": "pistol ammo, JHP (reloaded)" },
-    "proportional": { "price": 0.7, "damage": { "damage_type": "bullet", "amount": 0.9 }, "dispersion": 1.1 },
-    "extend": { "effects": [ "RECYCLED" ] }
+    "proportional": { "damage": { "damage_type": "bullet", "amount": 1.25, "armor_penetration": 0.001 } }
```
Comment (Coolthulhu) [inline data/mods/Generic_Guns/ammo/pistol.json:18]: Why `0.001`?
Does it not accept 0?

**Comment by Coolthulhu** [comment]
Alright, it seems that `proportional` doesn't like 0. This is stupid, but it's a problem in the code and not the PR.

**Comment by Lamandus** [comment]
Yes, proportional of 0 leads to 1.

**Pull request merged**